include/ddc/chunk.hpp

```diff
@@ -60,23 +61,29 @@ class Chunk<ElementType, DiscreteDomain<DDims...>>
 
     using reference = typename base_type::reference;
 
-    template <class, class>
+    template <class, class, class>
     friend class Chunk;
 
+private:
+    Allocator m_allocator;
+
 public:
     /// Empty Chunk
     Chunk() = default;
 
     /// Construct a Chunk on a domain with uninitialized values
-    explicit Chunk(mdomain_type const& domain)
-        : base_type(new (std::align_val_t(64)) value_type[domain.size()], domain)
+    explicit Chunk(mdomain_type const& domain, Allocator const& allocator = Allocator())
+        : m_allocator(allocator)
+        , base_type(std::allocator_traits<Allocator>::allocate(m_allocator, domain.size()), domain)
```

Comment:
I think we have UB here, the base_type should be initialized before m_allocator but `allocate` captures by lvalue& the Allocator.